neqo-http3/src/push_stream.rs

```diff
@@ -93,35 +93,43 @@ impl RecvStream for PushStream {
         loop {
             match &mut self.state {
                 PushStreamState::ReadPushId(id_reader) => {
-                    let push_id = id_reader.get_type(conn, self.stream_id);
-                    let fin = id_reader.fin();
-                    if fin {
-                        self.state = PushStreamState::Closed;
-                        return Ok(ReceiveOutput::NoOutput);
-                    }
-                    if let Some(p) = push_id {
-                        if self
-                            .push_handler
-                            .borrow_mut()
-                            .add_new_push_stream(p, self.stream_id)?
-                        {
-                            self.state = PushStreamState::ReadResponse {
-                                push_id: p,
-                                response: RecvMessage::new(
-                                    MessageType::Response,
-                                    self.stream_id,
-                                    Rc::clone(&self.qpack_decoder),
-                                    Box::new(RecvPushEvents::new(p, Rc::clone(&self.push_handler))),
-                                    None,
-                                ),
-                            };
-                        } else {
-                            mem::drop(conn.stream_stop_sending(
-                                self.stream_id,
-                                Error::HttpRequestCancelled.code(),
-                            ));
+                    let to_read = id_reader.min_remaining();
+                    let mut buf = vec![0; to_read];
+                    match conn.stream_recv(self.stream_id, &mut buf[..])? {
+                        (_, true) => {
                             self.state = PushStreamState::Closed;
-                            return Ok(ReceiveOutput::NoOutput);
+                            break Ok(ReceiveOutput::NoOutput);
```

Comment:
So if we get a push and the push ID is truncated, we don't close the connection?  That seems like it might be worth raising an error for.  I get that we ignore resets, but this is a FIN.